configutil/normalize_test.go

```diff
@@ -0,0 +1,534 @@
+		},
+		{
+			name:     "ipv6 URL RFC-5952 4.1 conformance leading zeroes",
+			address:  "https://[2001:0db8::0001]:8200",
+			expected: "https://[2001:db8::1]:8200",
+		},
+		{
+			name:     "ipv6 bracketed destination address with port RFC-5952 4.1 conformance leading zeroes",
+			address:  "username@[2001:0db8::0001]:8200",
+			expected: "username@[2001:db8::1]:8200",
+		},
+		{
+			name:     "ipv6 RFC-5952 4.2.2 conformance one 16-bit 0 field",
+			address:  "2001:db8:0:1:1:1:1:1",
+			expected: "2001:db8:0:1:1:1:1:1",
+		},
+		{
+			name:     "ipv6 URL RFC-5952 4.2.2 conformance one 16-bit 0 field",
+			address:  "https://[2001:db8:0:1:1:1:1:1]:8200",
+			expected: "https://[2001:db8:0:1:1:1:1:1]:8200",
+		},
+		{
+			name:     "ipv6 destination address with port RFC-5952 4.2.2 conformance one 16-bit 0 field",
+			address:  "username@[2001:db8:0:1:1:1:1:1]:8200",
+			expected: "username@[2001:db8:0:1:1:1:1:1]:8200",
+		},
+		{
+			name:     "ipv6 RFC-5952 4.2.3 conformance longest run of 0 bits shortened",
+			address:  "2001:0:0:1:0:0:0:1",
+			expected: "2001:0:0:1::1",
+		},
+		{
+			name:     "ipv6 URL RFC-5952 4.2.3 conformance longest run of 0 bits shortened",
+			address:  "https://[2001:0:0:1:0:0:0:1]:8200",
+			expected: "https://[2001:0:0:1::1]:8200",
+		},
+		{
+			name:     "ipv6 destination address with port RFC-5952 4.2.3 conformance longest run of 0 bits shortened",
+			address:  "username@[2001:0:0:1:0:0:0:1]:8200",
+			expected: "username@[2001:0:0:1::1]:8200",
+		},
+		{
+			name:     "ipv6 RFC-5952 4.2.3 conformance equal runs of 0 bits shortened",
+			address:  "2001:db8:0:0:1:0:0:1",
+			expected: "2001:db8::1:0:0:1",
+		},
+		{
+			name:     "ipv6 URL no port RFC-5952 4.2.3 conformance equal runs of 0 bits shortened",
+			address:  "https://[2001:db8:0:0:1:0:0:1]",
+			expected: "https://[2001:db8::1:0:0:1]",
+		},
+		{
+			name:     "ipv6 URL with port RFC-5952 4.2.3 conformance equal runs of 0 bits shortened",
+			address:  "https://[2001:db8:0:0:1:0:0:1]:8200",
+			expected: "https://[2001:db8::1:0:0:1]:8200",
+		},
+		{
+			name:     "ipv6 destination address with port RFC-5952 4.2.3 conformance equal runs of 0 bits shortened",
+			address:  "username@[2001:db8:0:0:1:0:0:1]:8200",
+			expected: "username@[2001:db8::1:0:0:1]:8200",
+		},
+		{
+			name:     "ipv6 RFC-5952 4.3 conformance downcase hex letters",
+			address:  "2001:DB8:AC3:FE4::1",
+			expected: "2001:db8:ac3:fe4::1",
+		},
+		{
+			name:     "ipv6 URL RFC-5952 4.3 conformance downcase hex letters",
+			address:  "https://[2001:DB8:AC3:FE4::1]:8200",
+			expected: "https://[2001:db8:ac3:fe4::1]:8200",
+		},
+		{
+			name:     "ipv6 destination address with port RFC-5952 4.3 conformance downcase hex letters",
+			address:  "username@[2001:DB8:AC3:FE4::1]:8200",
+			expected: "username@[2001:db8:ac3:fe4::1]:8200",
+		},
+		// NOTE: these tests are NOT conformant with the URI spec https://www.rfc-editor.org/rfc/rfc3986#section-3
```

Comment:
I assume these are in here for sake of review in comparing to the tests done in the Vault package and you will eventually take them out?

Reply:
yes